### ISSUE TITLE
Text colored in dialog

### ISSUE THREAD

**Issue by JSGUYOT** [frougon/pythondialog#12]
Hello,

I use pythondialog, thank's for your job :)

I try to color text in dialog, but is not work correctly in Python3
In old version (python2), it's worked nice, but in python3, it's almost work.
When i used this : "\Z5mytext\Zn", text is colored, but \ character apears in text.

Can you help me with this ?

Regards,

**Comment by frougon** [comment]
Hello,

When facing a problem like this, you should try it with the `dialog` program first, since pythondialog makes everything via `dialog`. `dialog` has a useful manual page where you can see that the `--colors` option is needed to interpret embedded “\Z” sequences in the dialog text.

So, here is a possibility:

```
#! /usr/bin/env python3

from dialog import Dialog
d = Dialog(dialog="dialog")
d.add_persistent_args(["--no-nl-expand", "--colors"])
d.msgbox(r"Foo \Z5bar\Zn baz")
```

Note that `--no-nl-expand` has nothing to do with this; I'm including it here as a hint because using it is generally a good idea for us Python users.

Regards

**Comment by frougon** [comment]
I hadn't seen the `\` part of your question, so I'm reopening the issue for now. My example doesn't show any backslash using `dialog` 1.3-20211214; can you provide a reproducible minimal example that does?

**Comment by JSGUYOT** [comment]
Hello Florent, 

thanks for your quick return :)

**For reproduce :** 

sudo pip3 list installed | grep dialog
pythondialog              3.5.3
sudo rpm -qa | grep dialog
dialog-1.3-13.20171209.el8.x86_64

```
#!/usr/bin/python3
# -*- coding: utf-8 -*-

from dialog import Dialog

global d
d = Dialog(dialog="dialog",autowidgetsize=True)
d.set_background_title("*** FOR TEST ***")
dTitle="Main Menu"
text = 'Select item'
choices = [
              ('1', 'Item 1'),
              ('2', 'Item 2'),
              ('3', 'Item 3'),
              ('', '\Z5-----------------------------\Zn'),
              ('4', 'Item 4')
            ]
code, tag = d.menu(text,choices=choices,title=dTitle,height=20,width=60,menu_height=8,extra_label='Exit',colors=True,ascii_lines=False,no_collapse=True,no_cancel=True,extra_button=True,no_mouse=True)
```

![image](https://user-images.githubusercontent.com/23457789/156325207-fbc9d523-79fa-424c-af9d-4c994b7b5276.png)

You can see colored text, but with the "\\" characters appear

Regards,

**Comment by frougon** [comment]
Okay,I see but running your example on my system doesn't exhibit the problem:

![pydlg_colors](https://user-images.githubusercontent.com/5341798/156341312-00626c83-cb74-4457-a2f0-58354d9816f3.png)

Again, this is with `dialog` 1.3-20211214 (tested in `xterm`). Your `dialog` program is a bit old; my suggestion would be to try with a more recent version.

For testing, you can compile `dialog` in-place: it doesn't *need* to be installed system-wide (of course, you must either put it early enough in `PATH` or use a full path in `d = Dialog(dialog="/path/to/dialog", ...)` so as to make sure the version you compiled will be used by pythondialog.

**Comment by JSGUYOT** [comment]
Hello Florent, 

Thanks for your time.

I'm work on RHEL8, and your dialog version isn't exist on EPEL repository for EL8 ... 
I not found source package with Makefile ..., and I'm not familiar with it without Makefile 

Regards,

**Comment by frougon** [comment]
Hello,

As I said, you  can build `dialog` [from source](https://invisible-island.net/dialog/dialog.html#download). Last time I built it, I used these commands:

```
./configure --prefix=/opt/dialog-1.3-20210621 --enable-nls --with-ncursesw
make
```

The `--prefix` is only useful in case I decide to run `make install` afterwards, but as said, this isn't needed.

Of course, this requires one to have already installed development packages for dependencies like `libncursesw` (`libncurses-dev` package on current Debian unstable, previously `libncursesw5-dev`).
